CounterpointGenerator/CounterpointGenerator/Generator.cs

```diff
@@ -40,21 +40,36 @@ private List<MelodyLine> GenerateCounterpointForNoteStack(MelodyLine m, Note? pr
 
         }
 
-        private List<Note> CounterpointForNote(Note n, Note prevN, Note preCtp)
+        private List<Note> CounterpointForNote(Note n, Note prevN, Note preCtp, int melodyLength, int generateCount)
         {
             List<Note> possibleNotes = GenerateNoteAtRegion(n);
-            List<Note> possibleNotesAfterRules = new List<Note>();
-            /*foreach(IRules r in Dictionary[input.userPreference])
+            // Possibilities, currentNote, position, length, previousCantus, previousCounterpoint
+            RuleInput ruleInput = new RuleInput(possibleNotes, n, generateCount, melodyLength, prevN, preCtp);
```

Comment:
Why do we want to create a new ruleInput object every time?

Reply:
Every field in the RuleInput is different every time CounterpointForNote is called. It gets called once, then the ruleapplier handles every rule sequentially, then CounterpointForNote returns the final list. The next recursion of GenerateCounterpointForNoteStack will have all new values for almost every field RuleInput wants